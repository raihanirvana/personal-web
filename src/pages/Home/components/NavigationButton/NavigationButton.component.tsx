import { Props } from './NavigationButton.component.type';

/**
 * function to render button for navigation
 *
 * @param {string} text text will visible
 * @returns {JSX.Element} return jsx element
 */
const NavigationButton = ({ text }: Props): JSX.Element => (
  <button className="border-white border-2 border-solid rounded-xl p-2 text-white w-50 hover:cursor-pointer text-2xl font-mono">
    {text}
  </button>
);

export default NavigationButton;
