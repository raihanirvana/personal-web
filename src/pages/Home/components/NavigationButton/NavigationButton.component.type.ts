export type Props = {
  text: string,
  navigate:string
}