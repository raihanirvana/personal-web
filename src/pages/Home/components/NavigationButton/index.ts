export { default } from './NavigationButton.component'