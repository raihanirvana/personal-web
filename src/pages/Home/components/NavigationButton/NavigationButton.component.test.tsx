import { render, screen, fireEvent } from '@testing-library/react';
import { MemoryRouter, Routes, Route } from 'react-router-dom';

import NavigationButton from './NavigationButton.component';

describe('NavigationButton', () => {
  it('must match snapshot', () => {
    const { asFragment } = render(<MemoryRouter><NavigationButton text='Halo' navigate='/resume' /></MemoryRouter>)

    expect(asFragment()).toMatchSnapshot()
  })

  it('must navigate when clicked', async () => {
    render(
    <MemoryRouter>
      <Routes>
      <Route path="/" element={<NavigationButton text='Resume' navigate='/resume' />} />
      <Route path="/resume" element={<div>Resume</div>} />
      </Routes>
    </MemoryRouter>)
    
    const linkButton = screen.getByRole('link', {name: 'Resume'})
    fireEvent.click(linkButton)

    expect(await screen.findByText('Resume')).toBeInTheDocument();
  })
});