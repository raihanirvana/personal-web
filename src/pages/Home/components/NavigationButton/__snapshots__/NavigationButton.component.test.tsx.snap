// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationButton must match snapshot 1`] = `
<DocumentFragment>
  <a
    class="border-white border-2 border-solid text-center rounded-xl p-2 text-white w-20 md:w-50 hover:cursor-pointer text-sm md:text-2xl font-mono hover:border-b-4 hover:border-blue-600 hover:border-t-0 hover:border-r-0 hover:border-l-1 hover:bg-black hover:rounded-lg"
    data-discover="true"
    href="/resume"
  >
    Halo
  </a>
</DocumentFragment>
`;
